svgcheck/run.py

```diff
@@ -107,12 +108,31 @@ def main():
         wp.color_threshold = options.grey_level
 
     if len(args) < 1:
-        source = os.getcwd() + "/stdin"
+        if sys.platform.lower().startswith("win") and sys.version_info < (3, 12):
+            # On Windows with Python < 3.12, can not reopen closed temp files.
+            sys.exit("Stdin is only supported on Windows with Python 3.12 and above.")
+
+        # delete_on_close was introduced in Python 3.12
+        delete_on_close = {"delete_on_close": False} if sys.version_info >= (3, 12) else {}
+
+        with tempfile.NamedTemporaryFile(mode="w+b", **delete_on_close) as tmp_file:
+            data = sys.stdin.buffer.read()
+            tmp_file.write(data)
+            if sys.version_info >= (3, 12):
+                tmp_file.close()
+            else:
+                # Prior to Python 3.12 file is deleted as soon as it's closed
+                tmp_file.seek(0)
+            source = tmp_file.name
+            process_svg(options, source)
     else:
         source = args[0]
+        process_svg(options, source)
         if not os.path.exists(source):
```

Comment:
Check the file exists before you process it?

Reply:
😨

Reply:
Good catch. I can add a test for that too.

Reply:
Fixed with https://github.com/ietf-tools/svgcheck/pull/56/commits/e83c0f6333d7f3879fd800cc3a0e826f47a45d16